src/main/java/capstone/mfslbackend/service/PlayerService.java

```diff
@@ -77,32 +77,29 @@ public Player createPlayer(PlayerResponse playerResponse, Long teamId) {
 
             if (!player.getPosition().equals(playerResponse.getPosition())) {
                 player.setPosition(playerResponse.getPosition());
-                playerRepository.save(player);
             }
-            if (playerResponse.getNumber()!=null && !playerResponse.getNumber().equals(player.getNumber()) ) {
+            if (playerResponse.getNumber() != null && !playerResponse.getNumber().equals(player.getNumber())) {
                 player.setNumber(playerResponse.getNumber());
-                playerRepository.save(player);
             }
-            if (!player.getTeam().getTeamId().equals(teamId)){
+            if (!player.getTeam().equals(team)) {
                 player.setTeam(team);
-                playerRepository.save(player);
             }
-            return player;
+
+            return playerRepository.save(player);
         }
-        Player player = new Player();
-        player.setPlayerId(playerResponse.getId());
-        player.setName(playerResponse.getName());
-        player.setUrl(playerResponse.getPhoto());
-        player.setPosition(playerResponse.getPosition());
-        player.setNumber(playerResponse.getNumber());
-        player.setTeam(team);
-        playerRepository.save(player);
-        return player;
+
+        Player player = new Player(
+                playerResponse.getId(),
+                playerResponse.getName(),
+                playerResponse.getPhoto(),
+                playerResponse.getPosition(),
+                playerResponse.getNumber(),
+                team
+        );
+        return playerRepository.save(player);
     }
 
     public Optional<Player> getPlayerById(Long playerId) {
-        Optional<Player> player = playerRepository.findById(playerId);
-        if (player.isEmpty()) log.warn("no player with id {} found", playerId);
-        return player;
+        return playerRepository.findById(playerId);
```

Comment:
Can you undo this. I did it like this so we would get a log (which is basically like a print statement) if ever a player is searched for in the db and isn't there

Reply:
changed it!